packages/amplify_datastore_plugin_interface/test/testData/Post.dart

```diff
@@ -13,22 +13,43 @@
 * permissions and limitations under the License.
 */
 
+// ignore_for_file: public_member_api_docs
+
+import 'ModelProvider.dart';
 import 'package:amplify_datastore_plugin_interface/amplify_datastore_plugin_interface.dart';
 import 'package:collection/collection.dart';
 import 'package:flutter/foundation.dart';
 
-import 'ModelProvider.dart';
-
 /** This is an auto generated class representing the Post type in your schema. */
 @immutable
 class Post extends Model {
   static const classType = const PostType();
   final String id;
-  final String title;
-  final int? rating;
-  final DateTime? created;
-  final Blog? blog;
-  final List<Comment>? comments;
+  final String? _title;
```

Comment:
question: Should `title` be required here? Is it required in `_internal`? Same question for `rating`? I'm just a little confused by different syntax for accessors below but I'm likely missing something.

Reply:
the codegen model fields are generated based on a graphql schema that unfortunately isn't shown in the file diff here.  The underlying fields (private _) are all nullable because we might create a model with just an id.  However, the field can be marked as required in the graphql schema.  Thus the public accessor makes a force cast "!"

Reply:
thanks